projects/neural_correlations/EXP1-Random/NeuCorr_Exp1.ipynb

```diff
@@ -173,79 +108,114 @@
     "                # Set range for plot appropriately!\n",
     "                plt.xlim(-0.05,0.1)\n",
     "                plt.xlabel(\"PCC\")\n",
     "                plt.ylabel(\"Frequency\")\n",
     "                plt.savefig(\"cellsHist_\" + str(ts))\n",
     "                plt.close()\n",
-    "                print \"++ Computing entropy ++\"\n",
     "                entropyX.append(ts)\n",
     "                entropyY.append(computeEntropy(subSpikeTrains))                \n",
-    "                \n",
-    "                print \"++ Analyzing correlations (whole columns) ++\"\n",
+    "\n",
+    "                #print \"++ Analyzing correlations (whole columns) ++\"\n",
     "                ### First the LARGE subsample of columns:\n",
     "                subSpikeTrains = subSampleWholeColumn(spikeTrains, colIndicesLarge, tm.getCellsPerColumn(), ts)\n",
     "                (corrMatrix, numNegPCC) = computePWCorrelations(subSpikeTrains, removeAutoCorr=True)\n",
     "                negPCCX_cols.append(s)\n",
     "                negPCCY_cols.append(numNegPCC)                \n",
-    "                print \"++ Generating histogram ++\"\n",
-    "                bins = 50\n",
+    "                #print \"++ Generating histogram ++\"\n",
+    "                bins = 300\n",
     "                plt.hist(corrMatrix.ravel(), bins, alpha=0.5)\n",
+    "                plt.xlim(-0.05,0.1)\n",
     "                plt.xlabel(\"PCC\")\n",
     "                plt.ylabel(\"Frequency\")\n",
     "                plt.savefig(\"colsHist_\" + str(ts))\n",
     "                plt.close()                 \n",
-    "                ### For the heatmap we consider the SMALL subsample in order to understand what's going on\n",
-    "                subSpikeTrains = subSampleWholeColumn(spikeTrains, colIndicesSmall, tm.getCellsPerColumn(), ts)\n",
-    "                (corrMatrix, numNegPCC) = computePWCorrelations(subSpikeTrains, removeAutoCorr=True)                \n",
-    "                print \"++ Generating heatmap ++\"\n",
-    "                plt.imshow(corrMatrix, cmap='spectral', interpolation='nearest')\n",
-    "                cb = plt.colorbar()\n",
-    "                cb.set_label('PCC')\n",
-    "                plt.savefig(\"colsHeatMap_\" + str(ts))\n",
-    "                plt.close()\n",
+    "                \n",
     "            ts += 1\n",
-    "    allSequences.append(sequence)\n",
-    "print \"***All sequences processed!***\""
+    "            \n",
+    "print \"*** DONE ***\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [],
+   "source": [
+    "sparsityTraceX = []\n",
+    "sparsityTraceY = []\n",
+    "for i in range(totalTS - 1000):\n",
+    "    sparsityTraceX.append(i)\n",
+    "    sparsityTraceY.append(np.mean(spikeCount[i:1000 + i]) / tm.numberOfCells())"
```

Comment:
sparsity can be efficiently computed using all cells during learning. No need to do it here on a small sample of the population.